boards/rp-pico/examples/pico_interpolator.rs

```diff
@@ -357,6 +390,89 @@ fn clamp(interp: &mut Interp1) -> bool {
     true
 }
 
+fn linear_interpolation(interp: &mut Interp0) -> bool {
+    let samples: [i16; 5] = [0, 10, -20, -1000, 500];
+    let uv_fractional_bits: u8 = 12;
+    // step is 1/4 in our fractional representation
+    let step: u32 = (1 << uv_fractional_bits) / 4;
+
+    // for lane 0
+    // shift and mask XXXX XXXX XXXX XXXX XXXX FFFF FFFF FFFF (accum 0)
+    // to 0000 0000 000X XXXX XXXX XXXX XXXX XXX0
+    // i.e. non fractional part times 2 (for uint16_t)
+    let config0 = LaneCtrl {
+        shift: uv_fractional_bits - 1,
+        mask_lsb: 1,
+        mask_msb: 32 - uv_fractional_bits,
+        add_raw: true,
+        blend: true,
+        ..LaneCtrl::new()
+    };
+    interp.get_lane0().set_ctrl(config0.encode());
+
+    // for lane 1
+    // shift XXXX XXXX XXXX XXXX XXXX FFFF FFFF FFFF (accum 0 via cross input)
+    // to 0000 XXXX XXXX XXXX XXXX FFFF FFFF FFFF
+
+    let config1 = LaneCtrl {
+        shift: uv_fractional_bits - 8,
+        signed: true,
+        cross_input: true,
+        ..LaneCtrl::new()
+    };
+    interp.get_lane1().set_ctrl(config1.encode());
+
+    interp.get_lane0().set_accum(0);
+    interp.set_base(samples.as_ptr() as u32);
+
+    let expected: [(i32, u32, i16, i16); 16] = [
+        (0, 0, 0, 10),
+        (2, 25, 0, 10),
+        (5, 50, 0, 10),
+        (7, 75, 0, 10),
+        (10, 0, 10, -20),
+        (2, 25, 10, -20),
+        (-5, 50, 10, -20),
+        (-13, 75, 10, -20),
+        (-20, 0, -20, -1000),
+        (-265, 25, -20, -1000),
+        (-510, 50, -20, -1000),
+        (-755, 75, -20, -1000),
+        (-1000, 0, -1000, 500),
+        (-625, 25, -1000, 500),
+        (-250, 50, -1000, 500),
+        (125, 75, -1000, 500),
+    ];
+    for (expected_peek, expected_percent, expected_pair_0, expected_pair_1) in expected {
+        // result2 = samples + (lane0 raw result)
+        // i.e. ptr to the first of two samples to blend between
+        let sample_pair: [i16; 2] = unsafe {
+            let sp: &[i16] = &(interp.peek() as *const [i16; 5]).read()[0..2];
```

Comment:
This example is really hard to follow. I wondered why this `unsafe` block is fine (and there should really be a comment explaining that). With some help from the datasheet I was able to infer how it works. I'd prefer if the examples were self-explaining and easy to understand.

Also I'm not sure if it's a great example, arithmetic on raw pointers isn't very usual in Rust programs. It might provide a performance benefit to let the interpolator do the whole calculation in one step, but that also should be justified by a comment, and ideally be verified by some benchmark or assembly analysis.

And, if I understand correctly what happens here (I'm not 100% sure!), this code has a bug and will trigger undefined behavior: `interp.peek() as *const [i16; 5]` is a pointer to an array `[i16; 5]`, but it points to some location within the original `samples` array. `read()` will read 5 `i16` values starting from that location, that is, it will read past the end of `samples`, which is not allowed. Even if then, only the first two of those values are actually used.